src/dataframe/dataframe.jl

```diff
@@ -904,6 +904,11 @@ function _deleteat!_helper(df::DataFrame, drop)
         return df
     end
 
+    selfdrop = findfirst(c -> c === drop, cols)
```

Comment:
you might want to use `Base.mightalias` instead - this would be even safer as would catch cases where user would be using e.g. views in the same fashion.

Reply:
@gustafsson - are you willing to work on improving this PR, or you are OK that I finalize it?

Reply:
Sure, I won't mind if you go ahead and finalize it

Good catch to use mightalias instead!